pkg/controller/migmigration/pod.go

```diff
@@ -108,195 +105,45 @@ func (t *Task) haveResticPodsStarted() (bool, error) {
-	for _, pod := range podList.Items {
-		if _, found := pod.Annotations[ResticPvBackupAnnotation]; !found {
-			continue
-		}
-		// Skip stage pods.
-		if _, found := pod.Labels[cLabel]; found {
-			continue
-		}
-		// Create
-		count++
-		newPod := t.buildStagePod(&pod)
-		err = client.Create(context.TODO(), newPod)
-		if err == nil {
-			log.Info(
-				"Stage pod created.",
-				"ns",
-				newPod.Namespace,
-				"name",
-				newPod.Name)
-			delete(pod.Annotations, ResticPvBackupAnnotation)
-			err = client.Update(context.TODO(), &pod)
-			if err != nil {
-				log.Trace(err)
-				return 0, err
-			}
-			continue
-		}
-		if !errors.IsAlreadyExists(err) {
-			log.Trace(err)
-			return 0, err
-		}
-	}
-
-	if count > 0 {
-		t.Owner.Status.SetCondition(migapi.Condition{
-			Type:     StagePodsCreated,
-			Status:   True,
-			Reason:   Created,
-			Category: Advisory,
-			Message:  "[] Stage pods created.",
-			Items:    []string{strconv.Itoa(count)},
-			Durable:  true,
-		})
-	}
-
-	return count, nil
-}
-
-// Ensure the stage pods are Running.
-func (t *Task) ensureStagePodsStarted() (bool, error) {
-	client, err := t.getSourceClient()
-	if err != nil {
-		log.Trace(err)
-		return false, err
-	}
-
-	podList := corev1.PodList{}
-	options := k8sclient.MatchingLabels(t.Owner.GetCorrelationLabels())
-	err = client.List(context.TODO(), options, &podList)
-	if err != nil {
-		return false, err
-	}
-	for _, pod := range podList.Items {
-		if pod.Status.Phase != corev1.PodRunning {
-			return false, nil
-		}
-	}
-
-	return true, nil
-}
-
 // Ensure the stage pods have been deleted.
 func (t *Task) ensureStagePodsDeleted() error {
-	clients, namespaceList, err := t.getBothClientsWithNamespaces()
+	client, err := t.getDestinationClient()
+	namespaceList := t.destinationNamespaces()
 	if err != nil {
 		log.Trace(err)
 		return err
 	}
 	cLabel, _ := t.Owner.GetCorrelationLabel()
-	for i, client := range clients {
-		podList := corev1.PodList{}
-		for _, ns := range namespaceList[i] {
-			options := k8sclient.InNamespace(ns)
-			err := client.List(context.TODO(), options, &podList)
-			if err != nil {
-				return err
+	podList := corev1.PodList{}
+	for _, ns := range namespaceList {
+		options := k8sclient.InNamespace(ns)
+		err := client.List(context.TODO(), options, &podList)
+		if err != nil {
+			return err
+		}
+		for _, pod := range podList.Items {
+			// Owned by ANY migration.
+			if pod.Labels == nil {
+				continue
 			}
-			for _, pod := range podList.Items {
-				// Owned by ANY migration.
-				if pod.Labels == nil {
-					continue
-				}
-				if _, found := pod.Labels[cLabel]; !found {
-					continue
-				}
-				// Delete
-				err := client.Delete(context.TODO(), &pod)
-				if err != nil && !errors.IsNotFound(err) {
-					log.Trace(err)
-					return err
-				}
-				log.Info(
-					"Stage pod deleted.",
-					"ns",
-					pod.Namespace,
-					"name",
-					pod.Name)
+			if _, found := pod.Labels[cLabel]; !found {
```

Comment:
The application pods on the destination will not have a _correlation label_ so I think this check can be removed. Instead update the `options` to filter on the StagePodLabel in addition to namespace.